pkg/imgpkg/util/tag.go

```diff
@@ -0,0 +1,20 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package util
+
+import (
+	"fmt"
+	regname "github.com/google/go-containerregistry/pkg/name"
+	regv1 "github.com/google/go-containerregistry/pkg/v1"
+)
+
+// BuildDefaultUploadTagRef Builds a tag from the digest Algorithm and Digest
+func BuildDefaultUploadTagRef(digest regv1.Hash, importRepo regname.Repository) (regname.Tag, error) {
```

Comment:
Since when we call this function everywhere we need to do a call to the `Digest` function maybe it would be better if we move that call to this function.
This would change the signature of the function to:
`func BuildDefaultUploadTagRef(digest WithDigest, importRepo regname.Repository) (regname.Tag, error)`
Create the following interface in `util`
```
type WithDigest interface {
   Digest() (regv1.Hash, error)
}
```
Not a fan of the name, maybe `Digestable` or something?

What do you think?